### PR TITLE
[dcaflare] Add profiling

### PR DESCRIPTION
**Pull request by davidor**
### What does this PR do?

Adds new options in the `datadog-cluster-agent flare` command to attach performance profiles: `--profile`, `--profile-blocking`, `--profile-blocking-rate`, `--profile-mutex`, `--profile-mutex-fraction`.

These are the same options that already exist in the command to generate agent flares.

```
datadog-cluster-agent flare --help
Collect a flare and send it to Datadog

Usage:
  datadog-cluster-agent flare [caseID] [flags]

Flags:
  -e, --email string                 Your email
  -h, --help                         help for flare
  -p, --profile int                  Add performance profiling data to the flare. It will collect a heap profile and a CPU profile for the amount of seconds passed to the flag, with a minimum of 30s (default -1)
  -B, --profile-blocking             Add goroutine blocking profile to the performance data in the flare
      --profile-blocking-rate int    Set the fraction of goroutine blocking events that are reported in the blocking profile (default 10000)
  -M, --profile-mutex                Add mutex profile to the performance data in the flare
      --profile-mutex-fraction int   Set the fraction of mutex contention events that are reported in the mutex profile (default 100)
  -s, --send                         Automatically send flare (don't prompt for confirmation)
```

I used #17231  by @mathieucolin as a starting point. I refactored the code a bit to adapt it to the new changes. I also extracted some functions that were duplicated, and fixed a bug (the code was using a settings client for the agent instead of the DCA).

### Describe how to test/QA your changes

- Deploy the agent on Kubernetes.
- Run `datadog-cluster-agent flare --profile 30` from the DCA pod.
- Check the profiles generated. For example, for the CPU one: `go tool pprof -http localhost:8080 datadog-cluster-agent-cpu.pprof`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by estherk15 — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
👍🏻

**Comment by davidor** [comment]
@L3n41c I've addressed your comments.

File: comp/core/flare/profiling.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	if _, err := c.Set(name, fmt.Sprint(value)); err != nil {
+		return nil, fmt.Errorf("failed to set %s to %v: %v", name, value, err)
+	}
+
+	return oldVal, nil
+}
+
+func resetRuntimeProfilingSettings(prev map[string]interface{}, settingsClient settings.Client) {
```
Comment (hush-hush) [inline comp/core/flare/profiling.go:65]: Could we maybe have a single function so set/reset settings ?

Something like `ExecWIthSetting(callback, setting...)` that would set the setting, exec the callback and reset everything back afterward ?

WDYT ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

**Comment by davidor** [comment]
@hush-hush I've rebased on top of main to fix the conflicts and also added a new commit to address your comments.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**